### PR TITLE
Gna conv2d decompose

### PR DESCRIPTION
**Pull request by sirzabek**
### Details:
 - add 2d convolution with 1d kernel to 1d convolution decomposition transform to allow it to run on GNA 2.0
 - add multiple test models covering all scenarios
 - add multiple test cases using all models and padding options.

### Tickets:
 - 48062

### CODE REVIEW

**Review by kbruniec — APPROVED** [review]

File: inference-engine/tests/functional/plugin/gna/pass_tests/conv2d_decomposition.cpp [hunk 1]
```diff
@@ -0,0 +1,251 @@
...
+};
+
+TEST_P(Conv2DDecomposeTest, CompareWithRefs) {
+    Run();
+}
+
+const std::vector<InferenceEngine::Precision> netPrecisions = {
+    InferenceEngine::Precision::FP32,
+    // TODO: tests with this setting are failing
+    //InferenceEngine::Precision::FP16
```
[the remark this answers is not in the document]
Comment (sirzabek) [reply inference-engine/tests/functional/plugin/gna/pass_tests/conv2d_decomposition.cpp:188]: Issue 55749.

File: inference-engine/tests/functional/plugin/gna/pass_tests/conv2d_decomposition.cpp [hunk 2]
```diff
@@ -0,0 +1,251 @@
...
+const std::vector<modelType> models = {
+    modelType::TranspConvTransp,
+    modelType::TranspConvBcastAddTransp,
+    modelType::TranspConvBcastAddActTransp,
+    //TODO: below two models are failing tests
+    //modelType::TranspConvTranspBcastAdd,
+    //modelType::TranspConvTranspBcastAddAct
+    //TODO: maxpool 2d is not supported yet by this transform
+    //modelType::TranspConvBcastAddMaxPoolTransp,
+    //modelType::TranspConvBcastAddMaxPoolActTransp,
```
[the remark this answers is not in the document]
Comment (sirzabek) [reply inference-engine/tests/functional/plugin/gna/pass_tests/conv2d_decomposition.cpp:214]: Issue 55749.

File: inference-engine/src/transformations/src/transformations/op_conversions/conv2d_decomposition.cpp [hunk 3]
```diff
@@ -0,0 +1,680 @@
...
+#include <ngraph/pattern/op/wrap_type.hpp>
+#include <ngraph_ops/type_relaxed.hpp>
+#include "itt.hpp"
+
+
+using namespace ngraph;
+using namespace op;
+
+#define GNA_MAX_1D_CONV_CHANNEL_COUNT 768
+#define GNA_MAX_PERMUTE_COL_COUNT 8
```
[the remark this answers is not in the document]
Comment (sirzabek) [reply inference-engine/src/transformations/src/transformations/op_conversions/conv2d_decomposition.cpp:23]: The ngraph api doesn't allow passing additional arguments, so these will need to remain inside the transformation.

File: inference-engine/tests_deprecated/unit/engines/gna/i16_quantisation_test.cpp [hunk 4]
```diff
@@ -160,34 +160,34 @@ TEST_F(I16QuantisationTest, OnlyAffineWithNanScaleFactorFails) {
...
 }
 
-TEST_F(I16QuantisationTest, EltwiseToMemory_ActivationInsertion) {
+TEST_F(I16QuantisationTest, DISABLED_EltwiseToMemory_ActivationInsertion) {
     assert_that().onInferModel(eltwiseToMemoryModel()).withGNAConfig(GNA_CONFIG_KEY(SCALE_FACTOR), 1.0f)
         .inNotCompactMode().gna().propagate_forward().called_with().pwl_inserted_into_nnet();
 }
 
 
-TEST_F(I16QuantisationTest, SplitFollowedByActivation_DummyDiagonalAffineInsertion) {
```
Comment (kbruniec) [inline inference-engine/tests_deprecated/unit/engines/gna/i16_quantisation_test.cpp:190]: The tests should be modified to re-enable.
@sirzabek consider to create an issue (or add to some existing issue to not forget.)

**Review by kbruniec — APPROVED** [review]

File: inference-engine/tests_deprecated/unit/engines/gna/i16_quantisation_test.cpp (hunk 4 above)
Comment (sirzabek) [reply inference-engine/tests_deprecated/unit/engines/gna/i16_quantisation_test.cpp:190]: Added to existing issue 55749.

**Pull request merged**